src/components/accordion/accordion-animated-panel.tsx

```diff
@@ -0,0 +1,53 @@
+import { animated, config, useSpring } from 'react-spring';
+import * as ReachAccordion from '@reach/accordion';
+import styled from 'styled-components';
+import { useAccordionItemContext } from '@reach/accordion';
+import { useDivHeight } from './use-div-height';
+import React from 'react';
+
+const AnimatedAccordionPanel = animated(ReachAccordion.AccordionPanel);
+const AnimatedPanelContainer = styled.div`
+  padding-bottom: 16px;
+`;
+/**
+ * Animated version of the AccordionPanel.
+ * Uses a custom hook `useDivHeight` to measure the height and then
+ * animated the height with `useSpring`.
+ *
+ * Based on forwardRef to forward the `ref` from the inner component `AnimatedAccordionPanel`
+ * to hide this animation layer implementation.
+ *
+ * Read this if forwardRef is unclear:
+ * https://reactjs.org/docs/forwarding-refs.html
+ *
+ * Based on:
+ * https://github.com/raunofreiberg/reach-ui/blob/6742f0cdcef05ba23077f63c21e3ede3bbfcae52/packages/accordion/examples/animated.example.js#L74-L89
+ * 1. We removed the forwardRef as we don't benefit from forwarding the ref as of now)
+ * 2. Added the aria-hidden flag based on this feedback: https://github.com/reach/reach-ui/pull/500#discussion_r403603183
+ */
+export const AccordionAnimatedPanel = ({ children }) => {
+  const { isExpanded } = useAccordionItemContext();
```

Comment:
This took years until I found it. It's in the documentation but virtually not described. It's a great technique to expose the expanded state of the accordion panel in any component that is interested in. I mean I know the context hooks but I just did not realize how flexible it can be used 👍 I'm glad I found this because without it you don't have access to the currennt state of the panel.

Reply:
The hook probably "only" works inside the `Accordion, as the accordion is creating the context for it's children. So not on `any component` but on any `child component`.